src/os/ObjectStore.h

```diff
@@ -761,9 +761,18 @@ class ObjectStore {
    *
    * @return iterator, null on error
    */
+  struct omap_iter_seek_t {
+    std::string seek_position;
+    enum {
+      LOWER_BOUND,
+      UPPER_BOUND
+    } seek_type = LOWER_BOUND;
+    static omap_iter_seek_t min_lower_bound() { return {}; }
+  };
   virtual ObjectMap::ObjectMapIterator get_omap_iterator(
     CollectionHandle &c,   ///< [in] collection
-    const ghobject_t &oid  ///< [in] object
+    const ghobject_t &oid, ///< [in] object
+    omap_iter_seek_t start_from = omap_iter_seek_t::min_lower_bound()  ///< [in] where the iterator should point to at the beginning
```

Comment:
Wouldn't be passing "const omap_iter_seek_t&" more straightforward/readable in terms of what's being copied during the call?